io/module_write_restart_netcdf.F90

```diff
@@ -0,0 +1,587 @@
+    type(ESMF_StaggerLoc)                :: staggerloc
+    type(ESMF_TypeKind_Flag)             :: attTypeKind
+    type(ESMF_Grid)                      :: wrtgrid
+    type(ESMF_Array)                     :: array
+    type(ESMF_DistGrid)                  :: distgrid
+
+    integer :: attCount
+    character(len=ESMF_MAXSTR) :: attName, fldName
+
+    integer :: varival
+    real(4) :: varr4val, dataMin, dataMax
+    real(4), allocatable, dimension(:) :: compress_err
+    real(8) :: varr8val
+    character(len=ESMF_MAXSTR) :: varcval
+
+    integer :: ncerr,ierr
+    integer :: ncid
+    integer :: oldMode
+    integer :: dimid
+    integer :: im_dimid, im_p1_dimid, jm_dimid, jm_p1_dimid, time_dimid
+    integer :: im_varid, im_p1_varid, jm_varid, jm_p1_varid, time_varid
+    integer, dimension(:), allocatable :: dimids_2d, dimids_3d
+    integer, dimension(:), allocatable :: varids, zaxis_dimids
+    logical shuffle
+
+    logical :: get_im_jm, found_im_jm
+    integer :: rank, deCount, localDeCount, dimCount, tileCount
+    integer :: my_tile, start_i, start_j
+    integer, dimension(:,:), allocatable :: minIndexPDe, maxIndexPDe
+    integer, dimension(:,:), allocatable :: minIndexPTile, maxIndexPTile
+    integer, dimension(:), allocatable :: deToTileMap, localDeToDeMap
+    logical :: do_io
+    integer :: par_access
+
+    real(ESMF_KIND_R8), allocatable  :: valueListr8(:)
+    logical :: isPresent, thereAreVerticals, is_restart_core, dynamics_restart_file
+    integer :: udimCount
+    character(80), allocatable :: udimList(:)
+    ! character(32), allocatable :: field_checksums(:)
+    character(32) :: axis_attr_name
+    character(32) :: field_checksum
+!
+    is_restart_core = .false.
+    if ( index(trim(filename),'fv_core.res') > 0 ) is_restart_core = .true.
+    tileCount = 0
+    my_tile = 0
+    start_i = -10000000
+    start_j = -10000000
+
+    par = use_parallel_netcdf
+    do_io = par .or. (mype==0)
+
+    call ESMF_FieldBundleGet(wrtfb, fieldCount=fieldCount, rc=rc); ESMF_ERR_RETURN(rc)
+
+    allocate(compress_err(fieldCount)); compress_err=-999.
+    allocate(fldlev(fieldCount)) ; fldlev = 0
+    allocate(fcstField(fieldCount))
+    allocate(varids(fieldCount))
+    allocate(zaxis_dimids(fieldCount))
+    ! allocate(field_checksums(fieldCount))
+
+    call ESMF_FieldBundleGet(wrtfb, fieldList=fcstField, grid=wrtGrid, &
+                             itemorderflag=ESMF_ITEMORDER_ADDORDER, &
+                             rc=rc); ESMF_ERR_RETURN(rc)
+
+    call ESMF_GridGet(wrtgrid, dimCount=gridDimCount, rc=rc); ESMF_ERR_RETURN(rc)
+
+    found_im_jm = .false.
+    do i=1,fieldCount
+       call ESMF_FieldGet(fcstField(i), dimCount=fieldDimCount, array=array, rc=rc); ESMF_ERR_RETURN(rc)
+       call ESMF_FieldGet(fcstField(i), name=fldName, rank=rank, typekind=typekind, rc=rc); ESMF_ERR_RETURN(rc)
+
+       if (fieldDimCount > 3) then
+          write(0,*)"write_netcdf: Only 2D and 3D fields are supported!"
```

Comment:
Maybe change to: if(mype==0) write(0,*) "ERROR: write_netcdf: Only 2D and 3D fields are supported!"